eventual-cassandra/src/test/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/ReplicatedCassandraTest.scala

```diff
@@ -1086,89 +1092,219 @@ class ReplicatedCassandraTest extends AnyFunSuite with Matchers {
     }
 
-    test(s"purge meta journal only, $suffix") {
-      val id      = "id"
-      val key     = Key(id, topic0)
-      val segment = segmentOfId.metaJournal(key)
-      val stateT = for {
-        _ <- journal.append(
-          key = key,
-          Partition.min,
-          Offset.unsafe(3),
-          timestamp   = timestamp0,
-          expireAfter = none,
-          events = Nel.of(
-            eventRecordOf(seqNr = SeqNr.unsafe(1), partitionOffset = PartitionOffset(Partition.min, Offset.unsafe(1))).event,
-            eventRecordOf(seqNr = SeqNr.unsafe(2), partitionOffset = PartitionOffset(Partition.min, Offset.unsafe(2))).event,
+    test(s"on purge of never truncated journal, purge all segments from start, $suffix") {
+      if (segmentSize == SegmentSize.max) assert(true, "cannot check for `next` segment on Int.MaxValue")
+      else {
+        val id                  = "id"
+        val key                 = Key(id = id, topic = topic0)
+        val segment             = segmentOfId.metaJournal(key)
+        val offset              = Offset.unsafe(2)
+        val segments            = 10
+        val baseSeqNr           = segmentSize.value * segments
+        val headPartitionOffset = PartitionOffset(Partition.min, offset)
+
+        val initial = State(
+          metaJournal = Map(
+            (
+              (topic0, segment),
+              Map(
+                (
+                  id,
+                  MetaJournalEntry(
+                    journalHead = JournalHead(
+                      partitionOffset = headPartitionOffset,
+                      segmentSize     = segmentSize,
+                      seqNr           = SeqNr.unsafe(baseSeqNr),
+                      deleteTo        = none, // instructs to clean all segments
+                    ),
+                    created = timestamp0,
+                    updated = timestamp0,
+                    origin  = origin.some,
+                  ),
+                ),
+              ),
+            ),
           ),
         )
-        _ <- journal.delete(key, Partition.min, Offset.unsafe(4), timestamp0, SeqNr.unsafe(2).toDeleteTo, origin.some)
-        _ <- journal.purge(key, Partition.min, Offset.unsafe(5), timestamp1)
-      } yield {}
 
-      val actual = stateT.run(State.empty)
-      val expected = State(
-        actions = List(
-          Action.Delete(key, segment),
-          Action.UpdateDeleteTo(
-            key,
-            segment,
-            PartitionOffset(Partition.min, Offset.unsafe(4)),
-            timestamp0,
-            SeqNr.unsafe(2).toDeleteTo,
+        val stateT = journal.purge(key, partitionOffset.partition, offset, timestamp1)
+
+        val expected = State(
+          actions = List(
+            Action.DeleteMetaJournal(key, segment),
+          ) ++ {
+            for {
+              i <- segments to (0, -1)
+            } yield Action.DeleteRecords(key, SegmentNr.of[Option](i.toLong).get)
```

Comment:
If you need pure computation while some `F[_]` is used - try `Id` from cats, no potential runtime exceptions like in `Option#get`

Reply:
I, shamelessly, copied it from other place without thinking. Adjusted, please recheck.